src/piplint/__init__.py

```diff
@@ -187,18 +188,20 @@ def valid_version(version, compare, r_version):
         print("%s%s%s\n" % (colour.WARNING,
                             "\n".join(sorted(unknown_reqs_with_versions)),
                             colour.ENDC))
+        if error_on_extras:
+            code = 2
```

Comment:
Would you also want to append an error into `errors` in this case as well so L194 becomes truthy? I don't personally case, just throwing it out there as a suggestion since you're coercing this from a warning into an error.